base/regex.jl

```diff
@@ -486,6 +496,20 @@ function findall(t::Union{AbstractString,AbstractPattern}, s::AbstractString; ov
     return found
 end
 
+function findall(t::T, s::T; overlap::Bool=false) where {T <: Vector{UInt8}}
```

Comment:
This prototype appears to normally be defined for:
```suggestion
function findall(t::Union{AbstractString, AbstractPattern, AbstractVector{<:Union{Int8,UInt8}}},
                 s::Union{AbstractString, AbstractPattern, AbstractVector{<:Union{Int8,UInt8}}},
                 ;
                 overlap::Bool=false)
```

which appears to me to avoid the need to copy the code.

If feasible, I think this may be good to move this to `strings/search.jl`, with the other similar cases (you may need to move `AbstractPattern` to the `strings/basic.jl` file too).

Reply:
In that case I'd make an internal function like _findall and only define the public one for type constraints?

I don't think I can fit both methods in one go right?

Reply:
I don't see an obvious reason you'd need to do that (I might be misreading something though), but it could be useful (for code reuse). It seems the current pattern would be to name it `_asearch`

Reply:
I mean if I want to avoid duplicating the function body?